### PR TITLE
Add column for player ID to 3 FBref functions

### PR DESCRIPTION
**Pull request by tonyelhabr**
Add `Player_Href` column to `fb_match_shooting()`, `fb_advanced_match_stats()`, `fb_league_stats(team_or_player = "player")`. This column can be used as a unique identifier for players and can be used to navigate to the correct player page.

### CODE REVIEW

File: tests/testthat/test-fbref.R
Comment (tonyelhabr) [inline tests/testthat/test-fbref.R:1]: mostly just spacing / formatting changes

File: tests/testthat/test-fbref.R [hunk 1]
```diff
@@ -381,7 +398,9 @@ Sys.sleep(3)
 test_that("fb_league_stats() for players works", {
   testthat::skip_on_cran()
   testthat::skip_on_ci()
-  expected_player_shooting_cols <- c("Rk", "Player", "Nation", "Pos", "Squad", "Age", "Born", "Mins_Per_90", "Gls", "Sh_Standard", "SoT_Standard", "SoT_percent_Standard", "Sh_per_90_Standard", "SoT_per_90_Standard", "G_per_Sh_Standard", "G_per_SoT_Standard", "Dist_Standard", "FK_Standard", "PK", "PKatt", "xG_Expected", "npxG_Expected", "npxG_per_Sh_Expected", "G_minus_xG_Expected", "np:G_minus_xG_Expected", "Matches", "url")
+
+  expected_player_shooting_cols <- c("Rk", "Player", "Player_Href", "Nation", "Pos", "Squad", "Age", "Born", "MP_Playing Time", "Starts_Playing Time", "Min_Playing Time", "Mins_Per_90_Playing Time", "Gls", "Ast", "G+A", "G_minus_PK", "PK", "PKatt", "CrdY", "CrdR", "xG_Expected", "npxG_Expected", "xAG_Expected", "npxG+xAG_Expected", "PrgC_Progression", "PrgP_Progression", "PrgR_Progression", "Gls_Per 90 Minutes", "Ast_Per 90 Minutes", "G+A_Per 90 Minutes", "G_minus_PK_Per 90 Minutes", "G+A_minus_PK_Per 90 Minutes", "xG_Per 90 Minutes", "xAG_Per 90 Minutes", "xG+xAG_Per 90 Minutes", "npxG_Per 90 Minutes", "npxG+xAG_Per 90 Minutes", "Matches", "url")
```
Comment (tonyelhabr) [inline tests/testthat/test-fbref.R:402]: looks like FBref changed some of the columns here? (G minus xG is no longer on the current page, for example) we may consider changing `.rename_fb_cols()`

File: vignettes/extract-fbref-data.Rmd [hunk 2]
```diff
@@ -592,7 +592,7 @@ Use the `fb_player_goal_logs()` function to get logs of "goals" (the default), "
 
 ```
 jwp_url <- "https://fbref.com/en/players/3515d404/James-Ward-Prowse"
-goal_log <- function(team_urls = jwp_url, goals_or_assists="both")
```
Comment (tonyelhabr) [inline vignettes/extract-fbref-data.Rmd:595]: drive by fix

File: R/get_advanced_match_stats.R [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+    xml2::xml_attr(player_elements, "href"),
+    xml2::xml_text(player_elements)
+  )
+  res <- dplyr::mutate(
+    df,
+    "Player_Href" = players[df$Player],
+    .after = "Player"
+  )
+  return(res)
+}
```
Comment (tonyelhabr) [inline R/get_advanced_match_stats.R:15]: this is the improvement made by this PR. other changes are refactors, moving functions in functions to the top-level of the script file (for easier debugging)

File: R/fb_league_stats.R [hunk 4]
```diff
@@ -100,7 +107,7 @@
 #' @examples
 #' \dontrun{
 #' try({
-#' fb_season_team_stats(
```
Comment (tonyelhabr) [inline R/fb_league_stats.R:103]: drive by fix

**Comment by JaseZiv** [comment]
This is awesome @tonyelhabr! Thanks for addressing this.

 Let me know when you're ready to move this out of WIP status.

File: tests/testthat/test-fbref.R [hunk 5]
```diff
@@ -381,19 +400,19 @@ Sys.sleep(3)
...
   testthat::skip_on_cran()
   testthat::skip_on_ci()
-  expected_player_shooting_cols <- c("Rk", "Player", "Nation", "Pos", "Squad", "Age", "Born", "Mins_Per_90", "Gls", "Sh_Standard", "SoT_Standard", "SoT_percent_Standard", "Sh_per_90_Standard", "SoT_per_90_Standard", "G_per_Sh_Standard", "G_per_SoT_Standard", "Dist_Standard", "FK_Standard", "PK", "PKatt", "xG_Expected", "npxG_Expected", "npxG_per_Sh_Expected", "G_minus_xG_Expected", "np:G_minus_xG_Expected", "Matches", "url")
+  expected_player_shooting_cols <- c("Rk", "Player", "Player_Href", "Nation", "Pos", "Squad", "Age", "Born", "Mins_Per_90", "Gls_Standard", "Sh_Standard", "SoT_Standard", "SoT_percent_Standard", "Sh_per_90_Standard", "SoT_per_90_Standard", "G_per_Sh_Standard", "G_per_SoT_Standard", "Dist_Standard", "FK_Standard", "PK_Standard", "PKatt_Standard", "xG_Expected", "npxG_Expected", "npxG_per_Sh_Expected", "G_minus_xG_Expected", "np:G_minus_xG_Expected", "Matches", "url")
   epl_player_shooting_22 <- fb_league_stats(
     country = "BRA",
     gender = "M",
     season_end_year = 2022,
     tier = "1st",
-    stat_type = "standard",
```
Comment (tonyelhabr) [inline tests/testthat/test-fbref.R:390]: i think they added a tab for shooting? (or something like that) this was very broken for `stat_type = "standard"`, but had very close to the `expected` columns with `stat_type = "shooting"`

**Comment by tonyelhabr** [comment]
> This is awesome @tonyelhabr! Thanks for addressing this.
> 
> Let me know when you're ready to move this out of WIP status.

ready for review! i think there were some tests (unrelated) that weren't passing earlier, and i didn't have time to check before logging off for the day, yesterday.

**Review by JaseZiv — APPROVED** [review]
lgtm.

Really appreciate it!

**Pull request merged**